src/Agent.Sdk/IBaseContext.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+
+namespace Agent.Sdk
+{
+    public interface IBaseContext : ITraceWriter
+    {
+        string GetVariableValueOrDefault(string variableName);
```

Comment:
Curious: What made you decide to go this route instead of keeping `Variables` exposed directly?

Reply:
Because IExecutionContext and AgentPluginTaskExecutionContext use two different types to store variables. The former uses an instance of Variables the later uses Dictionary<String,VariableValue>. It would be too much work to reconcile that because Variables needs a IHostContext.